ERCS/erc-xxxx.md

```diff
@@ -0,0 +1,147 @@
+---
+eip: xxxx
+title: Sustainable Collections
+description: A standard for managing NFTs with dynamic fees and donation-based engagement.
+author: Gustavo Lobo (@gflobo)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-12-04
+requires: 721
+---
+
+### Abstract  
+
+This EIP defines a standard for economically sustainable NFTs using features derived from the ERC721 standard. It integrates dynamic minting fees, role-based access control, and donation-based engagement. These mechanisms align tokenomics with principles of scarcity and value attribution while enabling creators to foster community engagement through contributions. Additionally, the standard supports optional features like token raffles to further incentivize community participation.
+
+---
+
+### Motivation  
+
+NFT systems often face issues like inflationary supply and lack of mechanisms to incentivize meaningful engagement between creators and contributors. This EIP addresses these gaps by introducing:  
+- **Dynamic Minting Fees**: To align token costs with user activity and ownership.  
+- **Donation-Based Engagement**: Encouraging contributions to creators.  
+- **Role-Based Access**: Empowering creators while ensuring transparent governance.  
+- **Optional Raffles**: Providing a mechanism for creators to reward contributors in a gamified manner.
+
+---
+
+### Specification  
+
+#### Core Functionalities  
+
+1. **Dynamic Minting Fees:**  
+   Minting fees increase dynamically based on the number of tokens owned by the user. This discourages hoarding and promotes equitable token distribution.  
+   - Formula:  
+     ```solidity
+     function mintFee() public view returns (uint256) {
+        if (hasRole(DEFAULT_ADMIN_ROLE, msg.sender)) return 0;
+        uint256 baseFee = mintBaseFee;
+        uint256 incrementPercentage = mintRateIncrementPercentage;
+        uint256 userTokenCount = balanceOf(msg.sender);
```

Comment:
If NFTs are transferrable, then a holder can just transfer to another account they control prior to minting.

Reply:
Thats right. I'm trying a new strategy for logarithmic reduction based on user mints stored in a mapping.